lib/elasticity/index_mapper.rb

```diff
@@ -152,8 +152,20 @@ def map_hit(hit)
 
         highlighted = @document_klass.new(highlighted_attrs)
       end
+      if @document_klass.config.subclasses.present?
+        Object.const_get(subclass_types[hit["_type"]]).new(attrs.merge(highlighted: highlighted))
```

Comment:
For example, the parent class would have a class method `document_types` that returns the hash for the parent class and all it's subclasses.